app/scripts/modules/cloudfoundry/src/pipeline/stages/runJob/CloudfoundryRunJobStageConfig.tsx

```diff
@@ -52,21 +53,15 @@ export class CloudfoundryRunJobStageConfig extends React.Component<
     });
   };
 
-  private commandUpdated = (event: React.ChangeEvent<HTMLInputElement>) => {
-    this.props.updateStageField({ command: event.target.value });
-  };
-
-  private JobNameUpdated = (event: React.ChangeEvent<HTMLInputElement>) => {
-    this.props.updateStageField({ jobName: event.target.value });
-  };
-
-  private targetUpdated = (target: string) => {
-    this.props.updateStageField({ target });
+  private stageFieldChanged = (fieldName: string, value: any) => {
+    set(this.props.stage, fieldName, value);
```

Comment:
we shouldn't be modifying property values directly any more should be using call backs such as `updateStageField`